### PR TITLE
added a new article on azure policies in eslz

### PR DESCRIPTION
**Pull request by Mahesh-MSFT**
**This PR fixes**

### CODE REVIEW

**Review by hjscherer — CHANGES_REQUESTED** [review]
The new article represneted in the file azpol.md includes a lot of details that are also covered in other parts of the doc. We try to avoid content replication to reduce mainainace efford.
Anything you would like to call-out beside general policies approach?

**Comment by Mahesh-MSFT** [comment]
Thanks for reviewing the pr.

The reason behind adding this article was to articulate consolidated value of azure policies in eslz. This is a big differentiator for eslz compared to landing zone frameworks from partners. Having a consolidated view of this differentiation will help customers make decision between eslz and partner landing zone offers objectively. I agree that some policies are discussed elsewhere in the documentation. However, that documentation is in the context of specific area (networking, management, etc.). It also *assumes* that customer/partner has made decision to use eslz. Proposed article comes much early in the process of customers wanting to choose between multiple landing zone frameworks.

File: docs/reference/azpol.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+## Enforce audit and log information collection
+
+Lack of auditing and diagnostics information at granular level can impact operational practices. Incomplete audit information makes it difficult to correlate logs from multiple Azure services and form a coherent debugging experience.
+
+It is desirable that once Azure services are provisioned, they provide detailed information about Azure platform they interact with.
+Such information can be broadly divided into logs and metrics. Each Azure service can be further categorized into its sub-components (e.g. An Azure Public IP resource has `DDoSProtectionNotifications`, `DDoSMitigationReports` and `DDoSMitigationFlowLogs` as its sub-components. Collecting diagnostic information at these sub-categories can greatly enhance auditing and debugging experience.
+
+ ESLZ deploys a custom Policy Initiative to enforce logs and metrics collection at a deeper level which helps enterprises gather logs and metrics per Azure Service as shown below.
+
+| Azure Service  | Log Categories           |Metrics        |
```
Comment (hjscherer) [inline docs/reference/azpol.md:37]: Would be great if we could replace this table with a general statement. That metrics are enabled accross all services

File: docs/reference/azpol.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+## Auto-provision Private Link/Endpoint with Private DNS Zone
+
+Azure Private Link and Azure Private Endpoint provide access to Azure Platform-as-a-service (PaaS) services using private IP addresses. However, Azure Private DNS Zone is needed for DNS record resolution. Creation of Azure Private Zones for every application that needs to access Azure PaaS services is a management and maintenance challenge.
+
+Azure Private DNS Zone Group helps is grouping the Private Link connections by Azure Services (blob, queue, table, sql, etc.) using an Azure Private Zone per service.
+
+Enterprises can create central Azure Private Zones and ESLZ custom policies will auto-provision connections between Private Link/Endpoint and Private DNS Zone for following Azure services.
+
+|Azure Service    |
```
Comment (hjscherer) [inline docs/reference/azpol.md:205]: We expect that this list is growing. Would suggest that we remove it.

**Review by hjscherer — CHANGES_REQUESTED** [review]

File: docs/reference/azpol.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+However, when these pilots/PoCs make transition to production-ready enterprise applications, their use of public IP addresses is sometimes overlooked.
+
+Production workloads using public IPs without proper security measures in place can increase security risks. Malicious actors can potentially use public IP as a gateway to launch an attack. Many enterprise compliance policies do not allow use of public IP just to avoid exposure to such security risks.
+
+There is a custom ESLZ policy on denying creation of public IP address which prevents pubic IP getting created in a scope targeted by policy. Enterprises can easily prevent Virtual Machines (VMs) getting created with public IP using this policy.
+
+Similarly there is a custom ESLZ Policy Initiative (also referred as PolicySet) which  helps enterprises prevent Azure services getting created with a public IP address in the first place. Following key Azure services are prevented ever getting created with a Public IP address.
+
```
Comment (hjscherer) [inline docs/reference/azpol.md:16]: Would recommend to review this table and make this statement "Following key Azure services are prevented ever getting created with a Public IP address." more generic.

**Review by hjscherer — CHANGES_REQUESTED** [review]

**Review by hjscherer — APPROVED** [review]

**Pull request merged**